digitaltwins/samples/device/javascript/pnpSampleDevice.js

```diff
@@ -0,0 +1,212 @@
+  }
+  case commandNameTurnOn: {
+    await sendCommandResponse(request, response, 200, 'turn On response');
+    break;
+  }
+  case commandNameTurnOff: {
+    await sendCommandResponse(request, response, 200, 'turn Off response');
+    break;
+  }
+  default: // Is there no way to register for an unknown method?
+    await sendCommandResponse(request, response, 404, 'unknown method');
+    break;
+  }
+};
+
+const sendCommandResponse = async (request, response, status, payload) => {
+  try {
+    await response.send(status, payload);
+    console.log('Response to method \'' + request.methodName +
+              '\' sent successfully.' );
+  } catch (err) {
+    console.error('An error ocurred when sending a method response:\n' +
+              err.toString());
+  }
+};
+
+const helperLogCommandRequest = (request) => {
+  console.log('Received command request for comand name \'' + request.methodName + '\'');
+
+  if (!!(request.payload)) {
+    console.log('The command request paylaod :');
+    console.log(request.payload);
+  };
+};
+
+const helperCreateReportedPropertiesPatch = (propertiesToReport, componentName, actualResponse) => {
+  const componentPart = componentPrefix + componentName;
+  const patch = { [componentPart]: {} };
+
+  Object.keys(propertiesToReport).forEach((propertyName) => {
+    const propertyContent = { value: propertiesToReport[propertyName] };
+    if (actualResponse) {
+      propertyContent.ac = actualResponse.code;
+      propertyContent.ad = actualResponse.description;
+      propertyContent.av = actualResponse.version;
+    }
+    patch[componentPart][propertyName] = propertyContent;
+  });
+  console.log('The following properties will be updated for component:' + componentName);
+  console.log(patch);
+  return patch;
+};
+
+const updateComponentReportedProperties = (deviceTwin, patch, componentName) => {
+  deviceTwin.properties.reported.update(patch, function (err) {
+    if (err) throw err;
+    console.log('Properties have been reported for component:' + componentName);
+  });
+};
+
+const helperAttachHandlerForDesiredPropertyPatches = (deviceTwin, componentName) => {
+  const prefixAndComponentName = componentPrefix + componentName;
+  const versionProperty = '$version';
+
+  deviceTwin.on('properties.desired.' + prefixAndComponentName, (delta) => {
+    Object.entries(delta).forEach(([propertyName, propertyValue]) => {
+      propertyUpdateHandler(deviceTwin, componentName, propertyName, null, propertyValue.value, deviceTwin.properties.desired[versionProperty]);
+    });
+  });
+};
+
+const helperAttachExitListener = async (deviceClient) => {
+  const standardInput = process.stdin;
+  standardInput.setEncoding('utf-8');
+  console.log('Please enter q or Q to exit sample.');
+  standardInput.on('data', (data) => {
+    if (data === 'q\n' || data === 'Q\n') {
+      console.log('Clearing intervals and exiting sample.');
+      clearInterval(intervalToken);
+      deviceClient.close();
+      process.exit();
```

Comment:
>process.exit(); [](start = 6, length = 15)

Don't know that you need this.

Reply:
if you are referring to the process.exit how do i quit application otherwise ?